src/Shipments/PriceCalculator.php

```diff
@@ -124,28 +142,29 @@ private function determineServiceRateForShipment(ShipmentInterface $shipment): S
         $serviceRates = $shipment->getService()->getServiceRates([
             'has_active_contract' => 'true',
             'contract'            => $shipment->getContract(),
-            'weight'              => $shipment->getPhysicalProperties()->getWeight(),
-            'volumetric_weight'   => $shipment->getPhysicalProperties()->getVolumetricWeight(),
+            'weight'              => $this->getBillableWeight($shipment),
+            'volume'              => $shipment->calculateVolumeInDm3(),
         ]);
 
-        $shipmentOptionIds = array_map(function (ServiceOptionInterface $serviceOption) {
-            return $serviceOption->getId();
-        }, $shipment->getServiceOptions());
+        $shipmentOptionIds = array_map(
+            fn (ServiceOptionInterface $serviceOption) => $serviceOption->getId(),
+            $shipment->getServiceOptions(),
+        );
 
         array_filter($serviceRates, function (ServiceRateInterface $serviceRate) use ($shipmentOptionIds) {
```

Comment:
What exactly is being done here? dont we have to store the outcome of `array_filter` in a new variable?

Reply:
Ehm, good point, I only refactored part of it, but it seems this code is not doing anything useful.

The purpose is to only return rates that support the already set options (if any) on that shipment. I'm going to fix this